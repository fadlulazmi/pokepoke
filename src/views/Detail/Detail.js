import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getDetail, catchPokemon } from '../../store/action'
import Button from '../../components/Button'
import defaultImage from '../../assets/ic-default-image.svg'
import './styles.scoped.css'


export default function Detail(props) {
  const dispatch = useDispatch()
  const { detailPokemon } = useSelector(state => state)
  const { id } = props.match.params
  const {
    sprites,
    name,
    base_experience,
    weight,
    height,
    types,
    moves
  } = detailPokemon || {}
  useEffect(() => {
    dispatch(getDetail(id))
  }, [dispatch, id])

  const getParaghraphContent = (arr, key) => {
    let content = '';
    arr.forEach((el, i) => {
      if(i < 10){
        if(i === arr.length-1){
          content += el[key].name
        } else {
          content += `${el[key].name}, `
        }
      }
    });
    
    if(content.length > 100){
      content = content.substr(0,100) + '...';
    }
    return content
  }

  return (
    <div className="detail">
      <img alt="id" src={sprites ? sprites.front_default : defaultImage}/>
      <p>{name && name.toUpperCase()}</p>
      <div>
        <p>
          <b>Base Experience</b> <br /> {base_experience} 
        </p>
        <p>
          <b>Weight</b> <br /> {weight} 
        </p>
        <p>
          <b>Height</b> <br /> {height} 
        </p>
        <p>
          <b>Types</b> <br /> {types && getParaghraphContent(types, 'type')} 
        </p>
        <p>
          <b>Moves</b> <br /> {moves && getParaghraphContent(moves, 'move')} 
        </p>
      </div>
      <div>
        <Button onClick={() => dispatch(catchPokemon(detailPokemon))} text="CATCH" />
      </div>
    </div>
  )
}
