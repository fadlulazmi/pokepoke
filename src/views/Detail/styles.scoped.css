.detail {
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-evenly;
  height: 100vh;
}

.detail > img {
  height: 30%;
  object-fit: contain;
}

.detail > p {
  font-size: 150%;
  font-weight: 700;
  margin: 0;
}

.detail > div {
  margin-top: 20px;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.detail > div > p {
  word-break: break-word;
  text-align: center;
  line-height: 35px;
  margin: 0;
  padding: 5px;
  border: lightgray 1px dashed;
  width: 146px;
  margin: 5px;
  font-size: 120%;
}

.detail > div > p:nth-child(5) {
  width: 317px;
}

.detail > div:last-child > button {
  margin: 0 10px;
  width: 150px;
  height: 50px;
}

.detail > div:last-child > a > button {
  margin: 0 10px;
  width: 150px;
  height: 50px;
}
