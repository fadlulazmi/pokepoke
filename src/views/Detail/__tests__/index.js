import Detail from '../Detail'
import ShallowRenderer from 'react-test-renderer/shallow';
import { useSelector } from 'react-redux';

describe('src/components/Detail', () => {
  test('should be rendered correctly', () => {
    useSelector.mockImplementationOnce(fn => {
      fn()
      return {
        detailPokemon: {}
      }
    })
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Detail match={{params: { id: '1' }}}/>);
    expect(tree).toMatchSnapshot();
  })
})
