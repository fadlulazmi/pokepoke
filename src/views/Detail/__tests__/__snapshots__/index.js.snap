// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Detail should be rendered correctly 1`] = `
<div
  className="detail"
>
  <img
    alt="id"
    src="ic-default-image.svg"
  />
  <p />
  <div>
    <p>
      <b>
        Base Experience
      </b>
       
      <br />
       
    </p>
    <p>
      <b>
        Weight
      </b>
       
      <br />
       
    </p>
    <p>
      <b>
        Height
      </b>
       
      <br />
       
    </p>
    <p>
      <b>
        Types
      </b>
       
      <br />
       
    </p>
    <p>
      <b>
        Moves
      </b>
       
      <br />
       
    </p>
  </div>
  <div>
    <Button
      onClick={[Function]}
      text="CATCH"
    />
  </div>
</div>
`;
